apps/internal/oauth/ops/authority/authority_ext_cachekey_test.go

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+package authority
+
+import (
+	"strings"
+	"testing"
+)
+
+func TestCacheExtKeyGenerator(t *testing.T) {
+	tests := []struct {
+		name     string
+		params   map[string]string
+		wantHash bool
+	}{
+		{
+			name:     "nil params",
+			params:   nil,
+			wantHash: false,
+		},
+		{
+			name:     "empty params",
+			params:   map[string]string{},
+			wantHash: false,
+		},
+		{
+			name: "single param",
+			params: map[string]string{
+				"param1": "value1",
+			},
+			wantHash: true,
+		},
+		{
+			name: "multiple params",
+			params: map[string]string{
+				"param1": "value1",
+				"param2": "value2",
+			},
+			wantHash: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ap := AuthParams{
+				CacheKeyComponents: tt.params,
+			}
+			hash := ap.CacheExtKeyGenerator()
+			if tt.wantHash && hash == "" {
+				t.Error("expected non-empty hash but got empty string")
+			}
+			if !tt.wantHash && hash != "" {
+				t.Errorf("expected empty hash but got %q", hash)
+			}
+		})
+	}
+}
+
+func TestCacheKeyComponentHashConsistency(t *testing.T) {
+	// Test that the same parameters produce the same hash
+	params1 := map[string]string{
+		"param1": "value1",
+		"param2": "value2",
+	}
+
+	params2 := map[string]string{
+		"param1": "value1",
+		"param2": "value2",
+	}
+
+	ap1 := AuthParams{CacheKeyComponents: params1}
+	ap2 := AuthParams{CacheKeyComponents: params2}
+
+	hash1 := ap1.CacheExtKeyGenerator()
+	hash2 := ap2.CacheExtKeyGenerator()
+
+	if hash1 != hash2 {
+		t.Errorf("expected same hash for same parameter keys, got %q and %q", hash1, hash2)
+	}
+}
+
+func TestCacheKeyComponentHashInConsistency(t *testing.T) {
+	// Test that different parameters produce different hashes
+	params1 := map[string]string{
+		"param1": "value1",
+	}
+
+	params2 := map[string]string{
+		"param2": "value2",
+	}
+
+	ap1 := AuthParams{CacheKeyComponents: params1}
+	ap2 := AuthParams{CacheKeyComponents: params2}
+
+	hash1 := ap1.CacheExtKeyGenerator()
+	hash2 := ap2.CacheExtKeyGenerator()
+
+	if hash1 == hash2 {
+		t.Errorf("expected different hashes for different parameter keys, both got %q", hash1)
+	}
+}
+
+func TestAppKeyWithCacheKeyComponent(t *testing.T) {
```

Comment:
this test will pass even if the implementation doesn't sort the keys

Reply:
No, map iteration order is undefined. You can only test probabilistically whether the AppKey method returns a stable value. Try adding a few more pairs to the test `params` and a loop around AppKey calls